src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2880,6 +2897,21 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
         }
     }
 
+    if (isAddTrackersFromURLEnabled() && !(hasMetadata && p.ti->priv()))
+    {
+        const auto maxTierIter = std::max_element(p.tracker_tiers.cbegin(), p.tracker_tiers.cend());
+        const int baseTier = (maxTierIter != p.tracker_tiers.cend()) ? (*maxTierIter + 1) : 0;
+
+        p.trackers.reserve(p.trackers.size() + static_cast<std::size_t>(m_additionalTrackerEntriesFromURL.size()));
+        p.tracker_tiers.reserve(p.trackers.size() + static_cast<std::size_t>(m_additionalTrackerEntriesFromURL.size()));
```

Comment:
copy paste error?
```suggestion
        p.tracker_tiers.reserve(p.tracker_tiers.size() + static_cast<std::size_t>(m_additionalTrackerEntriesFromURL.size()));
```

Reply:
This is copied from the code block directly above, though I believe it's intentional so that `p.trackers` and `p.tracker_tiers` are always the same size.